src/widgets/dnshole/TimerModal.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+  ActionIcon,
+  Button,
+  Flex,
+  Group,
+  Modal,
+  NumberInput,
+  NumberInputHandlers,
+  Stack,
+  Text,
+  rem,
+} from '@mantine/core';
+import { IconClockPause } from '@tabler/icons-react';
+import { useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface TimerModalProps {
+  toggleDns: any;
+  getDnsStatus(): any;
+  opened: boolean;
+  close(): any;
+  appId: string;
+}
+
+export function TimerModal({ toggleDns, getDnsStatus, opened, close, appId }: TimerModalProps) {
+  const [hours, setHours] = useState(0);
+  const [minutes, setMinutes] = useState(0);
+  const hoursHandlers = useRef<NumberInputHandlers>();
+  const minutesHandlers = useRef<NumberInputHandlers>();
+  const { t } = useTranslation('modules/dns-hole-controls');
+
+  return (
+    <Modal
+      withinPortal
+      radius="lg"
+      shadow="sm"
+      size="sm"
+      opened={opened}
+      onClose={close}
+      title={t('modules/dns-hole-controls:durationModal.title')}
+    >
+      <Flex direction="column" align="center" justify="center">
+        <Stack align="flex-end">
+          <Group spacing={5}>
+            <Text>{t('modules/dns-hole-controls:durationModal.hours')}</Text>
+            <ActionIcon
+              size={35}
+              variant="default"
+              onClick={() => hoursHandlers.current?.decrement()}
+            >
+              –
+            </ActionIcon>
+            <NumberInput
+              hideControls
+              value={hours}
+              onChange={(val) => setHours(Number(val))}
+              handlersRef={hoursHandlers}
+              max={23}
```

Comment:
Does the API stops us from using more than 24h? Why limit the hours since there isn't a "day" entry.

Reply:
We can use any time we want. Days, month.....
There is unlimited time for more then a day.

Reply:
Adding a day line would make it too much, so there's no reason to block the value at 23, let the user go higher if they want to.